providers/azure.go

```diff
@@ -216,18 +222,32 @@ func (p *AzureProvider) prepareRedeem(redirectURL, code string) (url.Values, err
 	return params, nil
 }
 
-// verifyTokenAndExtractEmail tries to extract email claim from either id_token or access token
+// verifyTokenAndExtractEmail tries to extract email from either id_token or access token
 // when oidc verifier is configured
 func (p *AzureProvider) verifyTokenAndExtractEmail(ctx context.Context, token string) (string, error) {
 	email := ""
 
+	oidcClaims, err := p.verifyTokenAndExtractClaims(ctx, token)
+	if err == nil && oidcClaims != nil {
+		email = oidcClaims.Email
+	}
+
+	return email, nil
+}
+
+// verifyTokenAndExtractClaims tries to extract oauth claims from either id_token or access token
+// when oidc verifier is configured
+func (p *AzureProvider) verifyTokenAndExtractClaims(ctx context.Context, token string) (*OIDCClaims, error) {
+	var oidcClaims *OIDCClaims
+	oidcClaims = nil
```

Comment:
The second line here is redundant, initialising a var like this means it is nil already